### PR TITLE
Upgrade docusaurus to 3.0.0

### PR DESCRIPTION
**Pull request by Hinton**
## Objective

Docusuarus should soon release v3.0.0. To evaluate how much effort this would be on our side I've gone through a quick upgrade.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,25 +30,33 @@
...
     "prettier": "3.0.3",
     "typescript": "5.2.2"
   },
+  "overrides": {
+    "docusaurus-lunr-search": {
+      "@docusaurus/core": "$@docusaurus/core",
+      "react": "$react",
+      "react-dom": "$react-dom"
+    }
+  },
```
Comment (Hinton) [inline package.json:59]: I've opened an upstream PR to docusaurus-lunr-search to avoid these.

**Comment by Hinton** [comment]
Noteworthy change:

```:::warning``` admonitions are now rendered differently. View https://ps-docusaurus-3.contributing-docs.pages.dev/getting-started/server/database/ef/ and compare it to https://contributing.bitwarden.com/getting-started/server/database/ef/

**Comment by withinfocus** [comment]
Do you want this considered for merge already, or is this to test beforehand? I'd think we can make some copytext improvements so you don't need to do the kind of escaping seen here.

**Comment by Hinton** [comment]
@withinfocus I think it's "working" as expected. We should figure out if we want to do anything about the warning visual changes. I'm also open for modifying the text to avoid escaping, but the escaping itself is pretty harmless.

**Review by withinfocus — APPROVED** [review]
I improved the copytext to avoid some escaping. Looks good to me! Do we wait for the Lunr improvement though?

File: docs/getting-started/server/sso/local.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ This uses
     docker-compose --profile idp up -d
     ```
 
-9.  You can test your user configuration by navigating to <http://localhost:8090/simplesaml>, then
+9.  You can test your user configuration by navigating to \<http://localhost:8090/simplesaml>, then
```
Comment (MGibson1) [inline docs/getting-started/server/sso/local.md:68]: ⚠️ only first angle bracket escaped. escape both or neither

File: docs/getting-started/server/sso/local.md [hunk 3]
```diff
@@ -99,7 +99,7 @@ however any currently authenticated users will have to log out for changes to th
 effect.
 
 To log out as a user, navigate to
-<http://localhost:8090/simplesaml/module.php/core/authenticate.php?as=example-userpass> and click
+\<http://localhost:8090/simplesaml/module.php/core/authenticate.php?as=example-userpass> and click
```
Comment (MGibson1) [inline docs/getting-started/server/sso/local.md:102]: ⚠️ only first angle bracket escaped. escape both or neither

**Review by MGibson1 — CHANGES_REQUESTED** [review]
It looks like you were rewriting to avoid un-named links, but misseed two

**Review by withinfocus — APPROVED** [review]

**Pull request merged**